src/Microsoft.DotNet.Build.Tasks/WriteItemsToJson.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class WriteItemsToJson : Task
+    {
+        [Required]
+        public string JsonFileName { get; set; }
+
+        [Required]
+        public ITaskItem[] Items { get; set; }
+
+        public override bool Execute()
+        {
+            if (!Directory.Exists(Path.GetDirectoryName(JsonFileName)))
+                Directory.CreateDirectory(Path.GetDirectoryName(JsonFileName));
+
+            JsonSerializer jsonSerializer = new JsonSerializer();
+            using (StreamWriter streamWriter = new StreamWriter(JsonFileName))
+            {
+                using (JsonTextWriter jsonWriter = new JsonTextWriter(streamWriter))
+                {
+                    jsonWriter.Formatting = Formatting.Indented;
+
+                    if (Items.Length > 1)
+                        jsonWriter.WriteStartArray();
+
+                    foreach (ITaskItem item in Items)
+                    {
+                        var customMd = item.CloneCustomMetadata();
+                        jsonWriter.WriteStartObject();
+                        foreach (var key in customMd.Keys)
+                        {
+                            var mdString = key.ToString();
+                            var mdValue = customMd[key].ToString();
+
+                            jsonWriter.WritePropertyName(mdString);
+
+                            if (mdValue.Length > 0 && mdValue[0] == '[' && mdValue[mdValue.Length - 1] == ']')
+                            {
+                                mdValue = mdValue.Substring(1, mdValue.Length - 2);
+                                jsonWriter.WriteStartArray();
+
+                                var parts = mdValue.Split(new char[] { ',' });
```

Comment:
This is going to cause a problem if the data contains commas, which seems like something it should handle.  It seems to be a sort of JSON encoded thing, since it's checking for [...], but doesn't handle quoting.

I'm a little confused as to what the format of these values are supposed to be.